packages/neon_http_client/test/interceptors/http_interceptor_test.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:http/http.dart' as http;
+import 'package:neon_http_client/src/interceptors/http_interceptor.dart';
+import 'package:test/test.dart';
+
+class _TestHttpInterceptor implements HttpInterceptor {
+  @override
+  http.BaseRequest interceptRequest({required http.BaseRequest request}) {
+    throw UnimplementedError();
+  }
+
+  @override
+  http.StreamedResponse interceptResponse({
+    required http.StreamedResponse response,
+    required Uri url,
+  }) {
+    throw UnimplementedError();
+  }
+
+  @override
+  bool shouldInterceptRequest(http.BaseRequest request) => throw UnimplementedError();
+
+  @override
+  bool shouldInterceptResponse(http.StreamedResponse response) => throw UnimplementedError();
+}
+
+void main() {
+  group('HttpInterceptor', () {
+    test('can be implemented', () {
+      expect(_TestHttpInterceptor(), isNotNull);
+    });
```

Comment:
What's the point of this?

Reply:
I like to have a test per dart file.
This way everything is explicitly covered.

If we where to remove the interceptors of this package (like moving them into the framework) the interface would still be verified to be implementable (not sealed ...).

Reply:
Hm ok :woman_shrugging: